src/storages/models/available_stores.rs

```diff
@@ -54,3 +67,21 @@ impl AvailableStores {
         }
     }
 }
+
+// TODO this should probably be a COW
+fn from_string_or_path(variable: &Option<String>, path: &Option<PathBuf>) -> Option<String> {
```

Comment:
As I see, it doesn't return Option, it returns string. Because otherwise it panics. 

Maybe you can replace Option with result or make it return just string.